sdk/servicebus/Azure.Messaging.ServiceBus/README.md

```diff
@@ -1,104 +1,361 @@
 
-The [get_thing](not-valid-link) method retrieves a Thing from the service. The `id` parameter is the unique ID of the Thing, not its "name" property.
+// create a message that we can send
+ServiceBusMessage message = new ServiceBusMessage(Encoding.Default.GetBytes("Hello world!"));
 
-```Python
-thing = client.get_thing(id)
+// send the message
+await sender.SendAsync(message);
+
+// get a receiver that we can use to receive and settle the message
+ServiceBusReceiver receiver = client.GetReceiver(queueName);
+
+// the received message is a different type as it contains some service set properties
+ServiceBusReceivedMessage receivedMessage = await receiver.ReceiveAsync();
+
+// store off the sequence number as we will need this to
+long sequenceNumber = receivedMessage.SequenceNumber;
+
+// defer the message, thereby preventing the message from being received again without using
+// the received deferred message API.
+await receiver.DeferAsync(receivedMessage);
+
+// receive the deferred message by specifying the service set sequence number of the original
+// received message
+ServiceBusReceivedMessage deferredMessage = await receiver.ReceiveDeferredMessageAsync(receivedMessage.SequenceNumber);
 ```
 
-### List the things
+### Dead letter a message
+
+```C# Snippet:ServiceBusDeadLetterMessage
+string connectionString = "<connection_string>";
+string queueName = "<queue_name>";
+// since ServiceBusClient implements IAsyncDisposable we create it with "await using"
+await using var client = new ServiceBusClient(connectionString);
+
+// get the sender
+ServiceBusSender sender = client.GetSender(queueName);
+
+// create a message that we can send
+ServiceBusMessage message = new ServiceBusMessage(Encoding.Default.GetBytes("Hello world!"));
 
-Use [list_things](not-valid-link) to get one or more Thing objects from the service. If there are no Things available, a `404` exception is thrown (see [Troubleshooting](#troubleshooting) for details on handling exceptions).
+// send the message
+await sender.SendAsync(message);
 
-```Python
-things = client.list_things()
+// get a receiver that we can use to receive and settle the message
+ServiceBusReceiver receiver = client.GetReceiver(queueName);
+
+// the received message is a different type as it contains some service set properties
+ServiceBusReceivedMessage receivedMessage = await receiver.ReceiveAsync();
+
+// store off the sequence number as we will need this to
+long sequenceNumber = receivedMessage.SequenceNumber;
+
+// deadletter the message, thereby preventing the message from being received again without receiving from the dead letter queue.
+await receiver.DeadLetterAsync(receivedMessage);
+
+// receive the dead lettered message with receiver scoped to the dead letter queue.
+ServiceBusReceiver dlqReceiver = client.GetDeadLetterReceiver(queueName);
+ServiceBusReceivedMessage dlqMessage = await dlqReceiver.ReceiveAsync();
+```
+
+### Using the Processor
```

Comment:
Since we support session features, can we provide one session example?

Reply:
I was thinking of leaving this out of the readme for now, but we can see what other languages are doing.

Reply:
I don't think we need to match the set of examples with other languages. I think if we are providing the features then we should add this in README.

Reply:
Yeah I meant - see if other languages that are implementing sessions are also including sessions in the README.

Reply:
Added.